addons/l10n_ro_edi_stock/models/l10n_ro_edi_stock_etransport_interface.py

```diff
@@ -0,0 +1,670 @@
+
+    @api.depends('picking_id.move_ids')
+    def _compute_l10n_ro_edi_stock_move_ids(self):
+        for interface in self:
+            interface.l10n_ro_edi_stock_move_ids = interface.picking_id.move_ids
+
+    @api.depends('picking_id.company_id')
+    def _compute_l10n_ro_edi_stock_company_id(self):
+        for interface in self:
+            interface.l10n_ro_edi_stock_company_id = interface.picking_id.company_id
+
+    @api.depends('picking_id.carrier_id')
+    def _compute_l10n_ro_edi_stock_carrier_id(self):
+        for interface in self:
+            interface.l10n_ro_edi_stock_carrier_id = interface.picking_id.carrier_id
+
+    @api.depends('l10n_ro_edi_stock_operation_type')
+    def _compute_l10n_ro_edi_stock_available_operation_scopes(self):
+        for interface in self:
+            allowed_scopes = [c for c, dummy in OPERATION_SCOPES]
+
+            if interface.l10n_ro_edi_stock_operation_type:
+                allowed_scopes = OPERATION_TYPE_TO_ALLOWED_SCOPE_CODES.get(interface.l10n_ro_edi_stock_operation_type, ("9999",))
+
+                if interface.l10n_ro_edi_stock_operation_scope not in allowed_scopes:
+                    interface.l10n_ro_edi_stock_operation_scope = None
+            elif interface.l10n_ro_edi_stock_operation_scope:
+                interface.l10n_ro_edi_stock_operation_scope = None
+
+            interface.l10n_ro_edi_stock_available_operation_scopes = ','.join(allowed_scopes)
+
+    @api.depends('l10n_ro_edi_stock_operation_type')
+    def _compute_available_location_types(self):
+        for interface in self:
+            interface.l10n_ro_edi_stock_available_start_loc_types = interface._get_available_location_types('start')
+            interface.l10n_ro_edi_stock_available_end_loc_types = interface._get_available_location_types('end')
+
+    @api.depends('l10n_ro_edi_stock_document_ids')
+    def _compute_current_document_data(self):
+        for interface in self:
+            document = interface._get_current_document()
+
+            interface.l10n_ro_edi_stock_state = document.state if document else False
+            interface.l10n_ro_edi_stock_document_message = document.message if document else False
+            interface.l10n_ro_edi_stock_document_load_id = document.load_id if document else False
+            interface.l10n_ro_edi_stock_document_uit = document.uit if document else False
+
+    ################################################################################
+    # Reset Data Methods
+    ################################################################################
+
+    def _reset_location_type(self):
+        for interface in self:
+            interface.l10n_ro_edi_stock_start_loc_type = 'location'
+            interface.l10n_ro_edi_stock_end_loc_type = 'location'
+
+    def _reset_location_data(self, location: str):
+        for interface in self:
+            setattr(interface, f'l10n_ro_edi_stock_{location}_bcp', False)
```

Comment:
Something that just crossed my mind, instead of `setattr` and `getattr`, can't you do this instead? (untested)
```py
interface[f'l10n_ro_edi_stock_{location}_bcp'] = False  # for setting

test = interface[f'l10n_ro_edi_stock_{location}_bcp']  # for getting

Reply:
I just tested it and it works fine, i'll push the change